SantasBag/SantasBag.DataAccess/Repositories/RewardsRepository.cs

```diff
@@ -0,0 +1,63 @@
+﻿using Microsoft.EntityFrameworkCore;
+using SantasBag.Core.Models;
+using SantasBug.Core.Abstractions;
+using SantasBug.DataAccess.Entities;
+
+namespace SantasBug.DataAccess.Repositories;
+
+public class RewardsRepository : IRewardsRepository
+{
+    private readonly SantasBagDbContext _context;
+
+    public RewardsRepository(SantasBagDbContext context)
+    {
+        _context = context;
+    }
+
+    public async Task<List<Reward>> Get(CancellationToken cancellationToken)
+    {
+        var rewardEntities = await _context.Rewards
+            .AsNoTracking()
+            .ToListAsync();
+        var rewards = rewardEntities
```

Comment:
First of all 'Get' method is used when we want to read some data. As I can see, you use Create method in LINQ Select, so it is not correct and developer who are going to use this method can be confused, because he doesn't expect such behavior.

So this method should be refactored and logic with creating can be made in business layer.

Reply:
Indeed, you helped me spot a bug that I missed at the current level of implementation.